metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/pattern_handler.py

```diff
@@ -324,6 +327,148 @@ def create_table_column_lineage(self, urn: str) -> List[ColumnLineageInfo]:
         return column_lineage
 
 
+class AmazonAthenaLineage(AbstractLineage):
+    """
+    Handles lineage extraction for Amazon Athena data sources in PowerBI.
+
+    Athena uses a three-level hierarchy: catalog.database.table
+    Example M-Query:
+        Source = AmazonAthena.Databases("us-east-1")
+        catalog = Source{[Name="AwsDataCatalog",Kind="Database"]}[Data]
+        database = catalog{[Name="my_database",Kind="Schema"]}[Data]
+        table = database{[Name="my_table",Kind="Table"]}[Data]
+
+    URN Format:
+        Generated URNs use database.table format (catalog is omitted) to match the
+        standalone Athena connector behavior. This ensures URN consistency across
+        different ingestion sources.
```

Comment:
what if "platform_instance" is set in the Athena source? should that be somehow considered here?

what other `AbstractLineage` implementations do regarding platform instance?